sql-parser/src/main/java/io/crate/sql/tree/AliasedRelation.java

```diff
@@ -82,7 +82,7 @@ public boolean equals(Object o) {
         if (!alias.equals(that.alias)) {
             return false;
         }
-        if (columnNames != null ? !columnNames.equals(that.columnNames) : that.columnNames != null) {
+        if (!columnNames.equals(that.columnNames)) {
```

Comment:
Could you add a `checkNotNull` check in the constructor above like for the other fields?